src/gennylib/include/gennylib/Node.hpp

```diff
@@ -455,10 +460,12 @@ class Node final {
     O to(Args&&... args) const {
         auto out = maybe<O, Args...>(std::forward<Args>(args)...);
         if (!out) {
-            BOOST_THROW_EXCEPTION(
-                InvalidKeyException("Tried to access node that doesn't exist.", this->key(), this));
+            auto x =
+                InvalidKeyException("Tried to access node that doesn't exist.", this->key(), this);
+            BOOST_LOG_TRIVIAL(fatal) << x.what();
```

Comment:
Unfortunately catch2 barfs on uncaught exceptions and doesn't print their `.what()` which makes debugging mis-configurations a real annoyance. This is infuriating but I see no easy way around it.  

However using the `boost::exception` machinery is a good idea so I added that in (but didn't fully use the boost error tags since they make the error messages icky and I don't have the energy to jump through boost's hoops to make them not-icky right now)